network/p2p/gossip/bloom.go

```diff
@@ -16,8 +17,7 @@ import (
 // anticipated at any moment, and a false positive probability of [targetFalsePositiveProbability]. If the
 // false positive probability exceeds [resetFalsePositiveProbability], the bloom filter will be reset.
 //
-// Invariant: The returned bloom filter is not safe to reset concurrently with
-// other operations. However, it is otherwise safe to access concurrently.
+// The returned bloom filter is safe for concurrent usage.
```

Comment:
Does this refactor encourage misuse of this bloom filter?

Specifically, it seems like this change does not expect for the bloom filter to be refilled atomically with its reset. This code certainly can be cleaned up... But adding this locking behavior doesn't seem useful (when using the bloom filter correctly).

If the bloom filter is accessed in a racy manner (Specifically `Marshal` being called concurrently with `ResetIfNeeded` followed by the expected `Add` calls), wouldn't we send a (potentially empty) bloom filter even though we may have a large number of items still in the set?

Reply:
I changed from a general callback to an `iter.Seq[Gossipable]` to reduce the scope to exactly what's needed.